From 68ec25c7b66ecfd5f2a383719d7688d09887b05a Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 12 Feb 2024 16:50:15 +0200
Subject: [PATCH] [SW-137122] gaudi3: erom rec: validate fit node hash

Added sha384 check function to verify the fit node hash.

Change-Id: I6e00cc0f6cd83f23b0955bf742ca6abdfbdb347a
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/arch/gaudi3/common/sha.c b/arch/gaudi3/common/sha.c
index 75c6772..3e89fed 100644
--- a/arch/gaudi3/common/sha.c
+++ b/arch/gaudi3/common/sha.c
@@ -23,3 +23,9 @@
 		sha384_update(&ctx, region[i].data, region[i].size);
 	sha384_finish(&ctx, output);
 }
+
+void sha_csum(const unsigned char *input, unsigned int ilen,
+		unsigned char *output)
+{
+	sha384_csum(input, ilen, output, 0);
+}
